src/CachePeers.h

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_SRC_CACHEPEERS_H
+#define SQUID_SRC_CACHEPEERS_H
+
+#include "CachePeer.h"
+#include "mem/PoolingAllocator.h"
+
+#include <memory>
+#include <vector>
+
+/// cache_peer configuration storage
+class CachePeers
+{
+public:
+    /// owns stored CachePeer objects
+    using Storage = std::vector< std::unique_ptr<CachePeer>, PoolingAllocator< std::unique_ptr<CachePeer> > >;
+
+    /// stores a being-configured cache_peer
+    void add(CachePeer *p) { storage.emplace_back(p); }
+
+    /// deletes a previously add()ed CachePeer object
+    void remove(CachePeer *);
+
+    /// the number of currently stored (i.e. added and not removed) cache_peers
+    size_t size() const { return storage.size(); }
+
+    /* peer iterators forming a sequence for C++ range-based for loop API */
+    using const_iterator = Storage::const_iterator;
+    const_iterator begin() const { return storage.cbegin(); }
+    const_iterator end() const { return storage.cend(); }
+
+    /// A CachePeer to query next when scanning all peer caches in hope to fetch
+    /// a remote cache hit. Never nil. \sa neighborsUdpPing()
+    /// \param iteration a 0-based index of a loop scanning all peers
+    CachePeer *nextPeerToPing(size_t iteration);
+
+private:
+    /// cache_peers in configuration/parsing order
+    Storage storage;
+
+    /// total number of completed peer scans by nextPeerToPing()-calling code
+    uint64_t peerPolls_ = 0;
```

Comment:
I am not sure I understand why we need this feature right now.
 * If it can be dropped please do,
OR
 * If we need to keep it, then this should be a `static` member of `CachePeer`.

With that changed this class can disappear entirely and the global config object provided by `CurrentCachePeers()` just be a `std::vector` of a `Pointer` type.

Reply:
> the `first_ping` variable it is effectively replacing

Aha. That was the detail I missed. Thank you for clarifying.